dialog/file.go

```diff
@@ -193,7 +199,7 @@ func (f *fileDialog) refreshDir(dir string) {
 		icons = append(icons, fi)
 	}
 	for _, file := range files {
-		if isHidden(file.Name(), dir) {
+		if !f.showHiddenCheck.Checked && isHidden(file.Name(), dir) {
```

Comment:
Reaching out of this context to read the value of a UI element seems like poor separation, this function takes a parameter for what directory to show, so it should probably have hidden logic similarly. 

Normally a method called with the same parameters should produce the same outcome